elfpy/data/convert_data.py

```diff
@@ -97,7 +99,160 @@ def fetch_contract_transactions_for_block(
+            # Two deltas, one for withdrawal shares, one for lp tokens
+            token_delta = _convert_scaled_value(-log["args"]["lpAmount"])
+            base_delta = _convert_scaled_value(log["args"]["baseAmount"])
+            wallet_deltas.append(
+                db_schema.WalletDelta(
+                    blockNumber=block_number,
+                    walletAddress=wallet_addr,
+                    baseTokenType="LP",
+                    tokenType="LP",
+                    tokenDelta=token_delta,
+                    baseDelta=base_delta,
+                )
+            )
+            token_delta = _convert_scaled_value(log["args"]["withdrawalShareAmount"])
+            base_delta = 0.0
+            wallet_deltas.append(
+                db_schema.WalletDelta(
+                    blockNumber=block_number,
+                    walletAddress=wallet_addr,
+                    baseTokenType="WITHDRAWAL_SHARE",
+                    tokenType="WITHDRAWAL_SHARE",
+                    tokenDelta=token_delta,
+                    baseDelta=base_delta,
+                )
+            )
+
+        elif log["event"] == "CloseLong":
+            wallet_addr = log["args"]["trader"]
+            token_delta = _convert_scaled_value(-log["args"]["bondAmount"])
+            base_delta = _convert_scaled_value(log["args"]["baseAmount"])
+            maturity_time = log["args"]["maturityTime"]
+            wallet_deltas.append(
+                db_schema.WalletDelta(
+                    blockNumber=block_number,
+                    walletAddress=wallet_addr,
+                    baseTokenType="LONG",
+                    tokenType="LONG-" + str(maturity_time),
+                    tokenDelta=token_delta,
+                    baseDelta=base_delta,
+                    maturityTime=maturity_time,
+                )
+            )
+
+        elif log["event"] == "CloseShort":
+            wallet_addr = log["args"]["trader"]
+            token_delta = _convert_scaled_value(-log["args"]["bondAmount"])
+            base_delta = _convert_scaled_value(log["args"]["baseAmount"])
+            maturity_time = log["args"]["maturityTime"]
+            wallet_deltas.append(
+                db_schema.WalletDelta(
+                    blockNumber=block_number,
+                    walletAddress=wallet_addr,
+                    baseTokenType="SHORT",
+                    tokenType="SHORT-" + str(maturity_time),
+                    tokenDelta=token_delta,
+                    baseDelta=base_delta,
+                    maturityTime=maturity_time,
+                )
+            )
+
+        elif log["event"] == "RedeemWithdrawalShares":
+            wallet_addr = log["args"]["provider"]
+            maturity_time = None
+            token_delta = _convert_scaled_value(-log["args"]["withdrawalShareAmount"])
+            base_delta = _convert_scaled_value(log["args"]["baseAmount"])
+            wallet_deltas.append(
+                db_schema.WalletDelta(
```

Comment:
WalletDelta is weird.  I think for the db case we should generalize it to just TokenDelta.  More atomic, probably easier queries.

Reply:
I see a case for doing a generic TokenDelta, but unfortunately this makes answering the question "how much base did you spend on this transaction" a bit harder to answer, especially when there's multiple transactions in the same block

Reply:
then I would add a transaction hash to the row as well.

Reply:
Thinking through more of this, there's def a way to map these wallet deltas back to transactions, but we'll need to put in a new reference to the transaction (probably back to the `Transactions` table) to match these deltas to a single transaction

Reply:
After talking with Sheng about it I think Matt has the right idea. Sheng's gonna do it in this PR.

Reply:
oops sorry I really meant for this to be a future consideration.

Reply:
Too late it's done